### PR TITLE
Upon `*blink`ing, nearby statues have a 0.1% chance of instantly snapping your neck

### PR DESCRIPTION
**Pull request by scriptis**
## About The Pull Request
Does what it says on the tin. This was a $20 (non-refundable) bounty provided by Pepper, regardless of whether it gets merged.

## Why It's Good For The Game
- Cool sci-fi references
- It's the little things that matter
- People who spam *blink should perish violently
- Also has the horror stinger sound from containment breach

## Changelog
:cl:
add: don't blink. don't blink not even for a second
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/emote.dm
Comment (distributivgesetz) [inline code/modules/mob/living/carbon/emote.dm:21]:
```suggestion
/datum/emote/living/carbon/blink/run_emote(mob/living/carbon/user, params, type_override, intentional)
	. = ..()
```
redundant check because this emote should only be available to carbons anyways

Comment (scriptis) [reply code/modules/mob/living/carbon/emote.dm:21]: i mean simplemobs can still *blink

they just can't get statue-murdered because they're neckless

**Comment by CCC23** [comment]
*blink
*blink
*blink
*blink
*CRUNCH

**Comment by Jackal-boop** [comment]
soul

**Comment by DATA-xPUNGED** [comment]
sci-fi reference? is peanut sci-fi now?

File: code/modules/mob/living/carbon/emote.dm
Comment (distributivgesetz) [reply code/modules/mob/living/carbon/emote.dm:21]: https://github.com/tgstation/tgstation/blob/ac446d8bad3596016217112a53fe91b8140dace2/code/modules/mob/living/carbon/emote.dm#L1-L2
you sure?

Comment (scriptis) [reply code/modules/mob/living/carbon/emote.dm:21]: huh yeah you're right

File: code/modules/mob/living/carbon/emote.dm
Comment (Kapu1178) [inline code/modules/mob/living/carbon/emote.dm:19]: put the prob before the view() call

File: code/modules/mob/living/carbon/emote.dm
Comment (Kapu1178) [inline code/modules/mob/living/carbon/emote.dm:30]: runtimes if the human has no head lol lmao dullahans

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by Ticklezcat** [comment]
> sci-fi reference? is peanut sci-fi now?

I'd say SCP counts as sci-fi, and more importantly Doctor Who is Definitely sci-fi.

File: code/modules/mob/living/carbon/emote.dm
Comment (scriptis) [reply code/modules/mob/living/carbon/emote.dm:19]: that'd make every statue in range dunk on you all at once--i'd prefer to have each statue individually maybe jump on you

File: code/modules/mob/living/carbon/emote.dm
Comment (scriptis) [reply code/modules/mob/living/carbon/emote.dm:30]: god why cant people just have heads

File: code/modules/mob/living/carbon/emote.dm
Comment (Watermelon914) [reply code/modules/mob/living/carbon/emote.dm:26]: do it anyways.
Just break early when it teleports behind a human once.

Comment (scriptis) [reply code/modules/mob/living/carbon/emote.dm:26]: ok

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
can you change this to only happen if there's a statue mob nearby, not an inanimate statue?
Normal statues shouldn't be sentient like this, but I don't mind this happening if a statue mob (aka the sort that can move when no one is looking at them) does this to you.

**Comment by MrMelbert** [comment]
this is gonna kill the shit out of me I *blink like 100 times a round, can we make this april fools or halloween exclusive please for the love of god

**Comment by scriptis** [comment]
the goal is soul lads--every statue's an angel

how about we lower it to a 0.1% chance

File: code/modules/mob/living/carbon/emote.dm
Comment (Watermelon914) [inline code/modules/mob/living/carbon/emote.dm:25]: View is a bad check to do here because it only takes into account the statues that the human can see. This means if a statue can see the human but the human can't see the statue, they will not get counted because you've used the human as the src instead of the location the human is on or the statue themselves.

**Comment by scriptis** [comment]
> can you change this to only happen if there's a statue mob nearby, not an inanimate statue?

to be perfectly clear: this is _meant_ to be for ordinary statues. the odds of it happening are just meant to be exceedingly low.

there are no statues on most maps by default with the exception of the goliath statue on lavaland, and a 1% (now 0.1%) proc chance for some doofus sculpting in the hallways sounds like fair game to me

soul

**Comment by scriptis** [comment]
given the current odds you'd need to blink 690 times in proximity of a statue for a 50% chance of it snapping your neck

**Comment by Ghommie** [comment]
1% or 0.1%, this is basically an unsolicited "Up Yours!" to whoever is that unlucky to emote nearby an otherwise innocuous, lifeless, structure and roll that slim chance. I'll abstain from judging, though personally I say that this makes a good addition for the cursed and the ones with omen component, not much for the rest.

**Comment by scriptis** [comment]
> 1% or 0.1%, this is basically an unsolicited "Up Yours!" to whoever is that unlucky to emote nearby an otherwise innocuous, lifeless, structure and roll that slim chance. I'll abstain from judging, though personally I say that this makes a good addition for the cursed and those with the omen component, not much for the rest.

I'll ask the pepper about whether this + halloween config is acceptable if the baseline is a step too far

File: code/modules/mob/living/carbon/emote.dm
Comment (carlarctg) [inline code/modules/mob/living/carbon/emote.dm:41]: nice

**Comment by carlarctg** [comment]
people should die to blinking it will be hilarious

File: sound/creatures/statue_horror_sting.ogg
Comment (MrMelbert) [inline sound/creatures/statue_horror_sting.ogg:1]: ...do we have the legal rights to use this?

**Comment by optimumtact** [comment]
>to be perfectly clear: this is meant to be for ordinary statues. the odds of it happening are just meant to be exceedingly low.

Then 0.1 is far too high.

**Comment by scriptis** [comment]
> Then 0.1 is far too high.

what's your appraisal? any number works for me

File: sound/creatures/statue_horror_sting.ogg
Comment (scriptis) [reply sound/creatures/statue_horror_sting.ogg:1]: let me double check and if not i'll swap it out with something CC0 licensed

Comment (DrDiasyl) [reply sound/creatures/statue_horror_sting.ogg:1]: SCP Containment Breach just used free horror sounds pack, as far as I know. You can hear this in some other horror games, so yeah.

Comment (scriptis) [reply sound/creatures/statue_horror_sting.ogg:1]: it's from https://www.youtube.com/watch?v=dRdpJq6nmWw, 2:08, CC BY 3.0

let me update attributions.txt

**Comment by DrDiasyl** [comment]
If this will get merged I know what I will do as Captain 😈

"WHO BLINKS THE MOST AMOUNT OF TIMES IN THIS ENCLOSED ROOM WITH A ASSISTANT STATUE WILL WIN THE SPARE ID!"
This will be _really fun._

**Comment by DrDiasyl** [comment]
Also I understand this is just a funny reference, but...

What about making some kind of traitor or wizard item which can curse statues of having a 100% chance of snapping a neck if you blink next to them?

**Comment by scriptis** [comment]
this PR has made me aware that i need to cycle my PGP keys

**Comment by CliffracerX** [comment]
This should *really* be exclusive to cursed statues and/or people with the Cursed quirk or Bad Omen smite applied.  Regular statues killing regular people once in a thousand blinks is still going to result in a lot of snapped necks and unwanted salt *and* likely ahelps alike considering most players aren't nolifing the github to check for things like this.

**Comment by DrDiasyl** [comment]
> This should *really* be exclusive to cursed statues and/or people with the Cursed quirk or Bad Omen smite applied.  Regular statues killing regular people once in a thousand blinks is still going to result in a lot of snapped necks and unwanted salt *and* likely ahelps alike considering most players aren't nolifing the github to check for things like this.

*Blink emote is not very popular, its not used that often. This decreases the chances of it appearing by even more, it can surely exist.

**Comment by Watermelon914** [comment]
I really think you should just do it on the statue mob, it would make things easier whilst allowing it to still be strong like this

**Comment by nfreader** [comment]
Please add a feedback key for this so we can track it on Statbus <3

**Comment by optimumtact** [comment]
and an achievement

**Comment by Jackal-boop** [comment]
so this is more rare than the arcade machine pulse rifle?

**Comment by moocowswag** [comment]
add something to the examine description of statues about how they should always be watched/dont blink/etc

stuff like this is fine when there is atleast some telltale as to what triggers it.

**Comment by san7890** [comment]
> so this is more rare than the arcade machine pulse rifle?

It's easier to assign blink to a keybind and spam it next to a statue you make, in fact the number of statues people spam in game make the likelihood of people getting into a "victory" condition as well as the ease it takes to accomplish said victory condition (you can surround yourself with like 8 statues and automatically increase your probability eight times).

**Comment by iwishforducks** [comment]
> > so this is more rare than the arcade machine pulse rifle?
> 
> It's easier to assign blink to a keybind and spam it next to a statue you make, in fact the number of statues people spam in game make the likelihood of people getting into a "victory" condition as well as the ease it takes to accomplish said victory condition (you can surround yourself with like 8 statues and automatically increase your probability eight times).

the chance runs when you blink, and then checks if there's statues nearby, rather than rolling a chance for each statue. so it only matters if a single statue is in your vision, not multiple.

**Comment by CPTANT** [comment]
I would very much prefer a much higher chance but only with statue mobs.

**Comment by dieamond13** [comment]
I don't get why people thing its too high of a chance, 1/1000 is really low especially for a fairly uncommon emote in a rare situation this would happen in

**Comment by scriptis** [comment]
I will add an achievement and a feedback key for this later this afternoon

**Comment by Ryll-Ryll** [comment]
Resident whimsy person here, why do we need a really threadbare, completely random, and deeply frustrating "easter egg" related to SCP-173 when we literally already have SCP-173 in game? Not to mention that I'm almost certain taking 200 brute to your head at once is going to cause it to dismember it entirely anyway (which maybe you do want that, idk what drove you to do this)

Funny little easter eggs and goofs are fine when they're not extremely stupidly balanced like this. Another maintainer can override this if they want, but I'm blocking this for now.

**Comment by optimumtact** [comment]
u dont even play

**Comment by Ryll-Ryll** [comment]
> u dont even play

That's how obvious this is stinky

**Comment by scriptis** [comment]
pepper says ur dumb :-1: :-1: :triumph: 

this may be re-opened (or speedmerged, at your option if you're based) by a more fun maintainer

**Comment by FlufflesTheDog** [comment]
In general dismemberment requires both bone and flesh wounds before even being considered, and I believe cannot happen on the same attack that causes said wounds. Because this is applied as blunt damage, it also doesn't seem to ever cause the required flesh wounds.

I tested this on local with some friends and had a blast with it. If you're blinking, you're likely near someone that can rush you off to medbay if you happen to win the 0.1% lottery, and the single broken bone with fairly high single-type damage made for a rather balanced revival, which was interesting but not super time consuming.

Incredibly soulful, 10/10

**Comment by iwishforducks** [comment]
NOOOOOOOOOOOOOOOOOOOOOOOO

**Pull request closed without merging**
